### ISSUE TITLE
[DataTable]: Keyboard navigation broken when using batch action version

### ISSUE THREAD

**Issue by janhassel** [carbon-design-system/carbon#5049]
## Environment

> Operating system
macOS 10.14.6

> Browser
Chrome 79
Firefox 72
Safari 12.1.2

## Detailed description

> What version of the Carbon Design System are you using?
carbon-components@10.9.1
carbon-components-react@7.9.1

When trying to use the batch action version of the DataTable, pressing <kbd>⇥ Tab</kbd> for the first time results in the toolbar being hidden (Chrome and Firefox; Safari highlights search). When pressing another time, the focus is still not visible (Chrome and Firefox; Safari loses focus completely and jumps to the next item on the page after the DataTable).
I need to press <kbd>⇥ Tab</kbd> three more times before Chrome and Firefox highlight the search.

## Steps to reproduce the issue

1. Open http://react.carbondesignsystem.com/?path=/story/datatable--with-batch-actions in Chrome or Firefox
2. Click once on the title "DataTable" to ensure keyboard focus is generally in the table
3. Press <kbd>⇥ Tab</kbd> repeatedly
